route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -236,85 +284,61 @@ namespace route_following_plugin
         }
         maneuver_msg.lane_change_maneuver.starting_lane_id = std::to_string(starting_lane_id);
         maneuver_msg.lane_change_maneuver.ending_lane_id = std::to_string(ending_lane_id);
-        current_time = maneuver_msg.lane_change_maneuver.end_time;
-        ROS_INFO_STREAM("Creating lane change start dist:"<<current_dist<<" end dist:"<<end_dist<<" Starting llt:"<<starting_lane_id<<" Ending llt:"<<ending_lane_id);
+        
+        ROS_INFO_STREAM("Creating lane change start dist:"<<start_dist<<" end dist:"<<end_dist<<" Starting llt:"<<starting_lane_id<<" Ending llt:"<<ending_lane_id);
         ROS_INFO_STREAM("Duration:"<<maneuver_msg.lane_change_maneuver.end_time.toSec() - maneuver_msg.lane_change_maneuver.start_time.toSec());
         return maneuver_msg;
     }
 
-    cav_msgs::Maneuver RouteFollowingPlugin::composeStopandWaitManeuverMessage(double current_dist, double end_dist, double current_speed, int start_lane_id, int end_lane_id, ros::Time& current_time, double end_time)
-    {
-        cav_msgs::Maneuver maneuver_msg;
-        maneuver_msg.type = cav_msgs::Maneuver::STOP_AND_WAIT;
-        maneuver_msg.stop_and_wait_maneuver.parameters.neogition_type = cav_msgs::ManeuverParameters::NO_NEGOTIATION;
-        maneuver_msg.stop_and_wait_maneuver.parameters.presence_vector = cav_msgs::ManeuverParameters::HAS_TACTICAL_PLUGIN;
-        maneuver_msg.stop_and_wait_maneuver.parameters.planning_tactical_plugin = "StopandWaitPlugin";
-        maneuver_msg.stop_and_wait_maneuver.parameters.planning_strategic_plugin = "RouteFollowingPlugin";
-        maneuver_msg.stop_and_wait_maneuver.start_dist = current_dist;
-        maneuver_msg.stop_and_wait_maneuver.start_speed = current_speed;
-        maneuver_msg.stop_and_wait_maneuver.end_dist = end_dist;
-        maneuver_msg.stop_and_wait_maneuver.start_time = current_time;
-        maneuver_msg.stop_and_wait_maneuver.starting_lane_id = std::to_string(start_lane_id);
-        maneuver_msg.stop_and_wait_maneuver.ending_lane_id = std::to_string(end_lane_id);
-        if(end_time < mvr_duration_){
-            end_time = mvr_duration_;
-        } 
-        maneuver_msg.stop_and_wait_maneuver.end_time = current_time + ros::Duration(end_time);
-        current_time = maneuver_msg.stop_and_wait_maneuver.end_time;
-        ROS_INFO_STREAM("Creating stop and wait start dist:"<<current_dist<<" end dist:"<<end_dist);
-        ROS_INFO_STREAM("Duration:"<<maneuver_msg.stop_and_wait_maneuver.end_time.toSec() - maneuver_msg.stop_and_wait_maneuver.start_time.toSec());
-        return maneuver_msg;
-    }
-    bool RouteFollowingPlugin::identifyLaneChange(lanelet::routing::LaneletRelations relations, int target_id)
+    bool RouteFollowingPlugin::isLaneChangeNeeded(lanelet::routing::LaneletRelations relations, int target_id)
     {
         for(auto& relation : relations)
         {
```

Comment:
Is SUCCESSOR really the correct check for identifying lane changes? It's been a while since I looked at the possible routing graph relations but couldn't a lane change also be a successor?

Reply:
I'm a little confused about this one, the description for Successor says " A (the only) direct, reachable successor. Not merging and not diverging." which does sound like it would not pick up a lane change, so our !Successor check should be good. But it does mean that at this point in the route, Successor should be an invalid check since there is no 'un-diverging' lanelet directly reachable

Reply:
@adev4a 
I looked into the source code and I think what you have will work correctly. Though there is a constraint on the input that the target lanelet be accessible from the other lanelet. If you had a non-accessible lanelet you would get the wrong result. I think this is fine given how you are using the method but you should probably document this constraint in the method comments